NotesRepository/Services/DirectoryService.cs

```diff
@@ -98,6 +229,23 @@ public bool RemoveDirectoriesSubdirectoriesAndNotesFromBinAndDbByDate(int daysOl
                 foreach (var directory in directories)
                 {
                     RemoveSubdirectoriesByDirectoryId(directory.DirectoryId);
+                    var directoryNotes = _nr.GetAllNotesForParticularDirectory(directory.DirectoryId);
```

Comment:
You should delete images also on notes, that have been moved 'in bulk' to the bin. Currently, you are removing images only from notes, that are under ANY directory, but there might be also a case, when the user moves a single note to the bin. The approach should be to check, whether each of `singleNotes` (7 lines of code above) has any images attached to it. When yes - then they should be obviously removed.